src/api/ThreadedComments.js

```diff
@@ -0,0 +1,340 @@
+        repliesCount,
+    }: {
+        errorCallback: (e: ElementsXhrError, code: string) => void,
+        fileId: string,
+        limit?: number,
+        marker?: string,
+        permissions: BoxItemPermission,
+        repliesCount?: number,
+        shouldFetchAll?: boolean,
+        successCallback: Function,
+    }): void {
+        this.errorCode = ERROR_CODE_FETCH_COMMENTS;
+        try {
+            this.checkApiCallValidity(PERMISSION_CAN_COMMENT, permissions, fileId);
+        } catch (e) {
+            errorCallback(e, this.errorCode);
+            return;
+        }
+
+        this.markerGet({
+            id: fileId,
+            successCallback,
+            errorCallback,
+            marker,
+            limit,
+            requestData: {
+                file_id: fileId,
+                ...(repliesCount ? { replies_count: repliesCount } : null),
+            },
+            shouldFetchAll,
+        });
+    }
+
+    /**
+     * @param {string} fileId - the file id
+     * @param {string} commentId - id of a Comment
+     * @param {BoxItemPermission} permissions - The known permissions of the comment
+     * @param {Function} successCallback - the success callback
+     * @param {Function} errorCallback - the error callback
+     */
+    getCommentReplies({
+        fileId,
+        commentId,
+        permissions,
+        successCallback,
+        errorCallback,
+    }: {
+        commentId: string,
+        errorCallback: (e: ElementsXhrError, code: string) => void,
+        fileId: string,
+        permissions: BoxItemPermission,
+        successCallback: Function,
+    }): void {
+        this.errorCode = ERROR_CODE_FETCH_REPLIES;
+
+        try {
+            this.checkApiCallValidity(PERMISSION_CAN_COMMENT, permissions, fileId);
+        } catch (e) {
+            errorCallback(e, this.errorCode);
+            return;
+        }
+
+        this.get({
+            id: fileId,
+            errorCallback,
+            successCallback,
+            url: this.getUrlWithRepliesForId(commentId),
+        });
+    }
+
+    /**
+     * @param {string} fileId - the file id
+     * @param {string} commentId - id of a Comment for which we createing Reply
+     * @param {BoxItemPermission} permissions - The known permissions of the comment
+     * @param {Function} successCallback - the success callback
+     * @param {Function} errorCallback - the error callback
+     * @param {string} message - message of the Reply
+     */
+    createCommentReply({
+        fileId,
+        commentId,
+        permissions,
+        successCallback,
+        errorCallback,
+        message,
+    }: {
+        commentId: string,
+        errorCallback: (e: ElementsXhrError, code: string) => void,
+        fileId: string,
+        message: string,
+        permissions: BoxItemPermission,
+        shouldFetchReplies?: boolean,
+        successCallback: Function,
+    }): void {
+        this.errorCode = ERROR_CODE_CREATE_REPLY;
```

Comment:
FYI: We may have some merge conflicts, as I believe these error codes are being used in the Annotations PR as well.